lib/matplotlib/colors.py

```diff
@@ -960,6 +960,208 @@ def scaled(self):
         return (self.vmin is not None and self.vmax is not None)
 
 
+class FuncNorm(Normalize):
+    """
+    Creates a normalizer using a custom function
+
+    The normalizer will be a function mapping the data values into colormap
+    values in the ``[0,1]`` range.
+    """
+
+    def __init__(self, f, finv=None, vmin=None, vmax=None, clip=False):
+        """
+        Specify the function to be used, and its inverse, as well as other
+        parameters to be passed to `Normalize`. The normalization will be
+        calculated as (f(x)-f(vmin))/(f(vmax)-f(vmin)).
+
+        Parameters
+        ----------
+        f : callable or string
+            Function to be used for the normalization receiving a single
+            parameter, compatible with scalar values and ndarrays.
+            Alternatively a string from the list ['linear', 'quadratic',
+            'cubic', 'x**{p}', 'sqrt', 'cbrt', 'root{p}(x)', 'log', 'log10',
+            'log2', 'log{p}(x)', 'log(x+{p}) 'log10(x+{p})', 'log{p}(x+{p})]
+            can be used, replacing 'p' by the corresponding value of the
+            parameter, when present.
+        finv : callable, optional
+            Inverse function of `f` that satisfies finv(f(x))==x.
+            Optional/ignored when `f` is a string.
+        vmin : float or None, optional
+            Value assigned to the lower limit of the colormap. If None, it
+            will be assigned to the minimum value of the data provided.
+            Default None.
+        vmax : float or None, optional
+            Value assigned to the upper limit of the colormap. If None, it
+            will be assigned to the maximum value of the data provided.
+            Default None.
+        clip : bool, optional
+            If True, any value below `vmin` will be clipped to `vmin`, and
+            any value above `vmax` will be clip to `vmin`. This effectively
```

Comment:
As far as I know, the default option is specified on the description, not in the specification, right?
From numpydoc:
> Optional keyword parameters have default values, which are displayed as part of the function signature. They can also be detailed in the description: